lib/cylc/scheduler.py

```diff
@@ -1582,14 +1589,21 @@ def run(self):
             self.main_loop_intervals.append(time() - tinit)
             # END MAIN LOOP
 
-    def update_state_summary(self):
-        """Update state summary."""
+    def update_data_structure(self):
+        """Update DB, UIS, Summary data elements"""
         updated_tasks = [
             t for t in self.pool.get_all_tasks() if t.state.is_updated]
         has_updated = self.is_updated or updated_tasks
         if has_updated:
+            # UI Server data update
+            # TODO: process the enitre pool once with self.is_updated
```

Comment:
s/enitre/entire (meh, not necessary, as it's a TODO, feel free to resolve this one :)